spec/services/stats/management_information/daily_report_query_spec.rb

```diff
@@ -392,35 +392,48 @@
       }
     end
 
-    context 'with claim state transitions' do
+    describe ':journey' do
       subject(:response) { described_class.call }
 
-      let(:transition_tos) { response.pluck(:journey).map { |el| el.pluck(:to) } }
+      let(:journey_tos) { response.pluck(:journey).map { |el| el.pluck(:to) } }
 
       it 'excludes state transitions to draft' do
         create(:advocate_final_claim, :allocated)
-        expect(transition_tos).to match_array([%w[submitted allocated]])
+        expect(journey_tos).to match_array([%w[submitted allocated]])
       end
 
       it 'excludes state transitions to archived_pending_delete' do
         create(:litigator_final_claim, :archived_pending_delete)
-        expect(transition_tos).to match_array([%w[submitted allocated authorised]])
+        expect(journey_tos).to match_array([%w[submitted allocated authorised]])
       end
 
       it 'excludes state transitions to archived_pending_review' do
         create(:litigator_hardship_claim, :archived_pending_review)
-        expect(transition_tos).to match_array([%w[submitted allocated authorised]])
+        expect(journey_tos).to match_array([%w[submitted allocated authorised]])
       end
 
-      # TODO: remove debug helper
-      # let(:time_info) do
-      #   { ruby_current_datetime: DateTime.current.iso8601,
-      #     ruby_now_datetime: DateTime.now.iso8601,
-      #     postgres_now: ActiveRecord::Base.connection.execute('select now()').to_a,
-      #     postgres_now_6_months_ago: ActiveRecord::Base.connection.execute("select (now() - '6 months'::interval)").to_a,
-      #     postgres_now_6_months_ago_cast: ActiveRecord::Base.connection.execute("select (now() - '6 months'::interval) at time zone 'utc' at time zone 'Europe/London'").to_a,
-      #     postgres_now_6_months_ago_trunc_cast: ActiveRecord::Base.connection.execute("select DATE_TRUNC('day', (now() - '6 months'::interval) at time zone 'utc' at time zone 'Europe/London')").to_a }
-      # end
+      # rubocop:disable RSpec/ExampleLength
+      it 'excludes deallocations and deallocated allocations' do
+        create(:advocate_final_claim, :allocated).tap do |claim|
+          claim.deallocate!
+          claim.allocate!
+          claim.deallocate!
+        end
+
+        expect(journey_tos).to eq([%w[submitted]])
+      end
+
+      it 'excludes deallocations and all but last allocation' do
+        create(:advocate_final_claim, :allocated).tap do |claim|
+          claim.deallocate!
+          claim.allocate!
+          claim.deallocate!
+          claim.allocate!
+        end
+
+        expect(journey_tos).to eq([%w[submitted allocated]])
+      end
+      # rubocop:enable RSpec/ExampleLength
```

Comment:
The long example length can be avoided with;

```ruby
    context 'with a claim allocated and deallocated twice' do
      before do
        create(:advocate_final_claim, :allocated).tap do |claim|
          claim.deallocate!
          claim.allocate!
          claim.deallocate!
        end
      end

      it { expect(journey_tos).to eq([%w[submitted]]) }
    end

    context 'with a claim allocated and deallocated twice and then allocated' do
      before do
        create(:advocate_final_claim, :allocated).tap do |claim|
          claim.deallocate!
          claim.allocate!
          claim.deallocate!
          claim.allocate!
        end

        it { expect(journey_tos).to eq([%w[submitted allocated]]) }
      end
    end
```

Reply:
i was trying to avoid using contexts too much as it looks neater in rspec output, but will have another look

Reply:
also, given complexity of the system i'd be tempted to bump the limit. These kind of rules are guides rather than must haves IMO

Reply:
rspec output from current is cleaner so would prefer to keep
```
    :journey
      excludes state transitions to archived_pending_review
      excludes state transitions to archived_pending_delete
      excludes deallocations and deallocated allocations
      excludes deallocations and all but last allocation
      excludes state transitions to draft
```